p4-16/psa/PSA.mdk

```diff
@@ -1497,6 +1499,59 @@ table.
 A PSA implementation must implement tables that have both a
 `psa_direct_counter` and `psa_direct_meter` property.
 
+### Table entry timeout notification {#sec-idle-timeout}
+
+PSA uses the `psa_idle_timeout` to enable a table implementation send
+notifications from the PSA device when a configurable time has passed since an
+entry was last matched. The property may take one of two values --
+`PSA_NO_TIMEOUT`, and `PSA_NOTIFY_CONTROL`. `PSA_NO_TIMEOUT` disables idle
+timeout support for the table. `PSA_NOTIFY_CONTROL` enables the notification. A
+PSA implementation will then generate an API for the control plane set
+time-to-live (TTL) values for table entries and if at any time during its
+lifetime, the table entry is not "hit" (&ie; not selected by any packet lookup)
+for a lapse of time greater or equal to its TTL, the device should generate a
+notification to the control plane.
+
+Example:
+```
+enum PSA_IdleTimeout_t {
+  PSA_NO_TIMEOUT,
+  PSA_NOTIFY_CONTROL
+}
+
+table t {
+  key = {};
+  actions = {};
+  psa_idle_timeout = PSA_NOTIFY_CONTROL;
```

Comment:
would be `PSA_IdleTimeout_t.PSA_NOTIFY_CONTROL`

Reply:
It might be nice to have at least one key and action with example names in there.  I know it is just an example, but I don't think you can actually add table entries if there are no key fields, so for such a table it would be useless to specify an idle timeout.

Reply:
done